lib/configuration.js

```diff
@@ -250,6 +250,27 @@ Configuration.prototype._gatherLocalConfiguration = function() {
   } else if (has(this._givenConfiguration, 'credentials')) {
     throw new Error('config.credentials must be a valid credentials object');
   }
+  this._checkAuthConfiguration();
+};
+/**
+ * The _checkAuthConfiguration function is responsible for determining whether
+ * or not a configuration instance, after having gathered configuration from
+ * environment and runtime, has credentials information enough to attempt
+ * authorization with the Stackdriver Errors API.
+ * @memberof Configuration
+ * @private
+ * @function _checkAuthConfiguration
+ * @returns {Undefined} - does not return anything
+ */
+Configuration.prototype._checkAuthConfiguration = function () {
+  if (isNull(this._key) && isNull(this._keyFilename) && 
+    isNull(this._credentials) && 
+    isEmpty(process.env.GOOGLE_APPLICATION_CREDENTIALS)) {
+      this._logger.error([
+        'Unable to find credential information on instance. This library will',
+        'be unable to communicate with the Stackdriver API to save errors.'
+      ].join(' '));
```

Comment:
Do we need to take any additional action here to disable the agent in the case where it cannot make authenticated requests?